ServerCore/Pages/Events/Import.cshtml.cs

```diff
@@ -130,25 +131,17 @@ public async Task<IActionResult> OnPostImportAsync()
                         destHint.Puzzle = puzzleCloneMap[sourceHint.Puzzle.ID];
                         _context.Hints.Add(destHint);
 
-                        foreach (Team team in _context.Teams.Where(t => t.Event == Event))
+                        foreach (Team team in destTeams)
                         {
                             _context.HintStatePerTeam.Add(new HintStatePerTeam() { Hint = destHint, TeamID = team.ID });
                         }
                     }
 
                     // PuzzleStatePerTeam
-                    foreach (Team team in _context.Teams.Where(t => t.Event == Event))
+                    foreach (Team team in destTeams)
                     {
                         int newPuzzleId = puzzleCloneMap[sourcePuzzle.ID].ID;
-                        bool hasPuzzleStatePerTeam = await (from pspt in _context.PuzzleStatePerTeam
```

Comment:
Note that the puzzle in puzzleCloneMap is always 100% new, see lines 87-91 above. That's why I believe this check can be removed. Also, there's no parallel existence check for hints in the block above...